ruby/ext/google/protobuf_c/message.c

```diff
@@ -329,6 +329,30 @@ VALUE Message_inspect(VALUE _self) {
   return str;
 }
 
+
+VALUE Message_to_h(VALUE _self) {
+  MessageHeader* self;
+  TypedData_Get_Struct(_self, MessageHeader, &Message_type, self);
+
+  VALUE hash = rb_hash_new();
+  rb_gc_register_address(&hash);
+
+  upb_msg_field_iter it;
+  for (upb_msg_field_begin(&it, self->descriptor->msgdef);
+       !upb_msg_field_done(&it);
+       upb_msg_field_next(&it)) {
+    const upb_fielddef* field = upb_msg_iter_field(&it);
+    VALUE msg_value = layout_get(self->descriptor->layout, Message_data(self), field);
+    VALUE msg_key   = rb_str_new2(upb_fielddef_name(field));
```

Comment:
I need to double check, but I think rails moved its 'active_record' methods to return symbols as the keys.  I would like to use symbols as they are more efficient and the rails change may give cover to make that change...?  But it would be a breaking change, so I'm leaving the old behavior for now (though the `#new` method requires symbols...)